src/pages/_app.js

```diff
@@ -8,9 +8,11 @@ import ApolloClient from 'apollo-boost';
 import { ApolloProvider } from '@apollo/react-hooks';
 
 import CssBaseline from '@material-ui/core/CssBaseline';
-import ThemeProvider from '@material-ui/styles/ThemeProvider';
-import StylesProvider from '@material-ui/styles/StylesProvider';
-import jssPreset from '@material-ui/styles/jssPreset';
+import {
+  jssPreset,
+  StylesProvider,
+  ThemeProvider
+} from '@material-ui/core/styles';
```

Comment:
@kilemensi  This defeats the entire purpose of what we are trying to do.

The `index.js` exports all this:
```
export * from './colorManipulator';
export { default as createMuiTheme, ThemeOptions, Theme, Direction } from './createMuiTheme';
export {
  default as createPalette,
  PaletteColorOptions,
  SimplePaletteColorOptions,
} from './createPalette';
export { default as createStyles } from './createStyles';
export { TypographyStyle, Variant as TypographyVariant } from './createTypography';
export { default as makeStyles } from './makeStyles';
export { default as responsiveFontSizes } from './responsiveFontSizes';
export { ComponentsPropsList } from './props';
export * from './transitions';
export { default as useTheme } from './useTheme';
export {
  default as withStyles,
  WithStyles,
  StyleRules,
  StyleRulesCallback,
  StyledComponentProps,
} from './withStyles';
export { default as withTheme, WithTheme } from './withTheme';
export { default as styled, ComponentCreator, StyledProps } from './styled';
export {
  createGenerateClassName,
  jssPreset,
  ServerStyleSheets,
  StylesProvider,
  ThemeProvider as MuiThemeProvider,
  ThemeProvider,
} from '@material-ui/styles';
```

Why are we against styles ?

Reply:
`import { makeStyles } from '@material-ui/core/styles';`

This will just bundle the rest of the exports thats why we need the absolute path.

Reply:
I'm **not** against `styles` @karimkawambwa  but I want us to not go off the documented path _unless_ we're sure there is a bug somewhere (and hopefully we've also reported it). Material UI is very clear about this, everywhere in their documentation they refer to `core/styles` rather than just `styles`: `styles` should only be used if you're not using `core`.

Using [Option 2](https://material-ui.com/guides/minimizing-bundle-size/#option-2):

1. Step 1: I've followed Step 1 _exactly_ in the configuration. There isn't anywhere that they're saying you need a separate transform for `styles`... If you've come across any issues here, please lets raise it rather than silently _fixing_ our config (Knowledge sharing and all).
2. Step 2. We need to convert all imports to `{ xxx }` syntax. I've done _manually_ for `makeStyles` related ones but Material UI provides a [script](https://github.com/mui-org/material-ui/blob/master/packages/material-ui-codemod/README.md#top-level-imports) for it... We can (should?) do it for the rest of imports.

But again, I'm not again `styles`.... If there are issues, we can & should switch to it. But only if _there are_ issues.

PS: Do we need this transform in HURUmap-UI as well?

Reply:
>Step 1: I've followed Step 1 exactly in the configuration. There isn't anywhere that they're saying you need a separate transform for styles... If you've come across any issues here, please lets raise it rather than silently fixing our config (Knowledge sharing and all).

@kilemensi just looking at the bundling, all the styles exports are definetly going to be bundled instead of just ServerStyleSheets you will get everything else.  But will look into it later. I'll just proceed with this.